File: Tareas Java/Impresion Arreglo VB/Impresion Arreglo VB/Form1.vb
Public Class Form1

    Dim formularioResultado As New FormResultado()

    Dim Arreglo As String(,)         'Arreglo para guardar los datos
    Dim Tabulaciones As Integer()    'Arreglo para guardar las tabulaciones de cada columna

    Dim filas, columnas, nroElementos As Integer
    Dim i, j, z As Integer 'Contadores

    Private Sub btnAgregarElemento_Click(sender As Object, e As EventArgs) Handles btnAgregarElemento.Click
        'Devuelve un arreglo con la tabulacion correspondiente a cada columnas

        nroElementos = filas * columnas

        If j = columnas Then
            j = 0
            i = i + 1
        End If

        'Guardamos el elemento y limpiamos el txtbox
        Arreglo(i, j) = txtElemento.Text
        txtElemento.Text = ""

        'Agregamos un nuevo margen de tabulacion si es más grande que el que
        'teníamos antes
        If Tabulaciones(j) < Arreglo(i, j).Length Then

            Tabulaciones(j) = Arreglo(i, j).Length

        End If

        j = j + 1
        z = z + 1

        If z = nroElementos Then

            Me.Hide()
            Array.Copy(Arreglo, formularioResultado.Arreglo, nroElementos)
            Array.Copy(Tabulaciones, formularioResultado.Tabulaciones, columnas)
            formularioResultado.Show()

        End If
    End Sub



    Private Sub btnSiguiente_Click(sender As Object, e As EventArgs) Handles btnSiguiente.Click
        If txtColumnas.Text = "" Or txtFilas.Text = "" Then

            'MsgBox();

        Else
            formularioResultado.filas = Convert.ToInt32(txtFilas.Text)
            formularioResultado.columnas = Convert.ToInt32(txtColumnas.Text)

            'Habilitamos el txt para recibir los elementos
            txtElemento.Enabled = True
            btnAgregarElemento.Enabled = True

            'Deshabilitamos los otros txtBoxs
            txtColumnas.Enabled = False
            txtFilas.Enabled = False
            btnSiguiente.Enabled = False

            filas = Convert.ToInt32(txtFilas.Text)
            columnas = Convert.ToInt32(txtColumnas.Text)

            ReDim Arreglo(filas - 1, columnas - 1)
            ReDim formularioResultado.Arreglo(filas - 1, columnas - 1)
            ReDim Tabulaciones(columnas - 1)
            ReDim formularioResultado.Tabulaciones(columnas - 1)

            z = 0

            'Llenamos todos los espacios del arreglo con 0
            For i = 0 To columnas - 1

                Tabulaciones(i) = 0

            Next
        End If


        j = 0
        i = 0
    End Sub
End Class

File: Tareas Java/Impresion Arreglo/Impresion Arreglo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Impresion_Arreglo;

namespace Impresion_Arreglo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        FormResultado formularioResultado = new FormResultado();

        String[,] Arreglo;      //Arreglo para guardar los datos
        int[] Tabulaciones;    //Arreglo para guardar las tabulaciones de cada columna

        int filas, columnas, nroElementos;
        int i, j, z; //Contadores
        

        private void button1_Click(object sender, EventArgs e)
        {

            if (txtColumnas.Text == "" || txtFilas.Text == "")
            {
                //MsgBox();
            }
            else
            {
                formularioResultado.filas = Convert.ToInt32(txtFilas.Text);
                formularioResultado.columnas = Convert.ToInt32(txtColumnas.Text);

                //Habilitamos el txt para recibir los elementos
                txtElemento.Enabled = true;

                //Deshabilitamos los otros txtBoxs
                txtColumnas.Enabled = false;
                txtFilas.Enabled = false;

                filas = Convert.ToInt32(txtFilas.Text);
                columnas = Convert.ToInt32(txtColumnas.Text);

                Arreglo = new String[filas, columnas];
                formularioResultado.Arreglo = new String[filas, columnas];
                Tabulaciones = new int[columnas];
                formularioResultado.Tabulaciones = new int[columnas];

                z = 0;
                //Llenamos todos los espacios del arreglo con 0
                for (i = 0; i < columnas; i++)
                {

                    Tabulaciones[i] = 0;

                }
            }

            i = j = 0;
        }

        
        

        private void button2_Click(object sender, EventArgs e)
        {
            //Devuelve un arreglo con la tabulacion correspondiente a cada columnas

            

            nroElementos = filas * columnas;

            if(j == columnas)
            {
                j = 0;
                i++;
            }


            //Guardamos el elemento y limpiamos el txtbox
            Arreglo[i, j] = txtElemento.Text;
            txtElemento.Text = "";

            //Agregamos un nuevo margen de tabulacion si es más grande que el que
            //teníamos antes
            if (Tabulaciones[j] < Arreglo[i, j].Length ){

                Tabulaciones[j] = Arreglo[i, j].Length;

            }

            j++;
            z++;

            if(z == nroElementos)
            {
                this.Hide();
                Array.Copy(Arreglo, formularioResultado.Arreglo, nroElementos);
                Array.Copy(Tabulaciones, formularioResultado.Tabulaciones, columnas);
                formularioResultado.Show();
            }
        }
    }
}
